src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1729,25 +1729,47 @@ bool isRefEscape
                 isRefEscape,
                 ignoreArglistRefKinds: true, // https://github.com/dotnet/roslyn/issues/63325: for compatibility with C#10 implementation.
                 argsAndParamsAll);
-            foreach (var argAndParam in argsAndParamsAll)
-            {
-                var argument = argAndParam.Argument;
-                uint argEscape = argAndParam.IsRefEscape ?
-                    GetRefEscape(argument, scopeOfTheContainingExpression) :
-                    GetValEscape(argument, scopeOfTheContainingExpression);
 
-                escapeScope = Math.Max(escapeScope, argEscape);
-                if (escapeScope >= scopeOfTheContainingExpression)
+            var returnsWritableRefToRefStruct = ReturnsRefToRefStruct(symbol);
+            foreach (var (param, argument, _, isArgumentRefEscape) in argsAndParamsAll)
+            {
+                // SPEC:
+                // If `M()` does return ref-to-ref-struct, the *safe-to-escape* is the same as the *safe-to-escape* of all arguments which are ref-to-ref-struct. It is an error if there are multiple arguments with different *safe-to-escape* because of *method arguments must match*.
```

Comment:
>It is an error if there are multiple arguments with different *safe-to-escape* because of *method arguments must match*. [](http://example.com/codeflow?start=162&length=120)

nit: skip second sentence?

Reply:
We felt this was more clear to include. It's meant to explain why we are still calculating the narrowest scope here (`Math.Max`) instead of requiring equal scopes--you shouldn't worry about the case where the scopes actually differ, since an error will be reported elsewhere.